### PR TITLE
[v2.7] Add K8s January patches

### PR DESCRIPTION
**Pull request by rayandas**
Related issue: https://github.com/rancher/rancher/issues/40210
This PR also includes template changes for ingress-nginx v1.5.1. Related issue: https://github.com/rancher/rancher/issues/40224 (depends on: https://github.com/rancher/ingress-nginx/pull/119)

### CODE REVIEW

File: rke/templates/nginx-v1.5.1-rancher1.go
Comment (kinarashah) [inline rke/templates/nginx-v1.5.1-rancher1.go]: Nginx v1.3 and 1.5 changed rbac permissions for a few groups that I don't see here, were they intentionally removed for some reason? (diff [here](https://github.com/kubernetes/ingress-nginx/commit/e222b74b9e1055e34259eab0281974af77281df6#diff-ef998dfe3ccaeba0d958bd51690eb731b39eade775395aa88328e376b76bba09)) 

we'd need to confirm others, but from release notes we do need for `coordination.k8s.io/leases` particularly: 
```
Starting with this release, we will need permissions on the coordination.k8s.io/leases resource for leaderelection lock
```

File: rke/templates/nginx-v1.5.1-rancher1.go
Comment (kinarashah) [inline rke/templates/nginx-v1.5.1-rancher1.go]: Same comment around rules update for `Role` would also apply here for `ClusterRole`

File: rke/templates/nginx-v1.5.1-rancher1.go
Comment (kinarashah) [inline rke/templates/nginx-v1.5.1-rancher1.go]: Unfortunately looks like 1.5.1 changed this again from `ingress-controller-leader` to `ingress-nginx-leader`. We depend our logic on if `DefaultIngressClass` is enabled or not but we'll need to confirm if this change is required for us or not.

**Review by kinarashah — CHANGES_REQUESTED** [review]
All rancher specific changes from old template are present in the new one, but some from upstream are missing or need to be revisited. 
I'd come back to re-reviewing this PR after 
- we tag a new `rancher/nginx-*` `1.5.1-rancher1` tag 
- test it locally once with RKE standalone.

**Comment by rayandas** [comment]
@kinarashah Alright, I updated the ingress-nginx PR, after a review I will merge that and then test the image with this templates. Also I will try to add what's missing as you said.

File: rke/templates/nginx-v1.5.1-rancher1.go
Comment (rayandas) [reply rke/templates/nginx-v1.5.1-rancher1.go]: Updated the template

**Comment by rayandas** [comment]
@kinarashah I updated the template, keeping `ingress-controller-leader` as it is. Added the KDM data `https://raw.githubusercontent.com/rayandas/kontainer-driver-metadata/rayandas-k8s-jan-update/data/data.json` in Rancher 2.7.1 and created two clusters with K8s `1.24.10` and `1.23.16` nginx ingress controller pods are running.
nginx
nginx123
Screenshot from 2023-01-31 18-02-05

File: rke/templates/templates.go
Comment (snasovich) [inline rke/templates/templates.go]: Ranges need to be updated here as we want `nginxIngressv120Rancher2` to still be used for 1.24.0 through 1.24.9. It's fine to have `>=1.24.10-rancher1-1": nginxIngressv151Rancher2,` as the only 1.25 version we're introducing is on newer version of Ingress.

**Review by snasovich — DISMISSED** [review]
One issue to address, otherwise LGTM. Feel free to merge after addressing it and getting @kinarashah 's approval.

File: rke/templates/templates.go
Comment (rayandas) [reply rke/templates/templates.go]: @snasovich Updated the PR with your suggestions.

Comment (snasovich) [reply rke/templates/templates.go]: @rayandas , thank you, but I think there is a misunderstanding here. We still want `v1.23.16-rancher1-1` to use new Ingress, I meant that we need to expand ranges in a way that only `v1.23.16-rancher1-1` (and later in 1.23.x) and `v1.24.10-rancher1-1` (and later) use 1.5.1 while previous versions stay on `nginxIngressv120Rancher2`.

Comment (rayandas) [reply rke/templates/templates.go]: Ah! got it. I think this is the correct versioning @snasovich Updated the PR
```
">=1.23.6-rancher1-1 <1.23.16-rancher1-1":  nginxIngressv120Rancher2,
">=1.23.16-rancher1-1 <1.24.0-rancher1-1":  nginxIngressv151Rancher2,
">=1.24.2-rancher1-1 <1.24.9-rancher1-1":   nginxIngressv120Rancher2,
">=1.24.10-rancher1-1":                     nginxIngressv151Rancher2,

**Review by snasovich — APPROVED** [review]

File: rke/templates/templates.go
Comment (snasovich) [reply rke/templates/templates.go]: Thank you, @rayandas . LGTM.

**Review by kinarashah — APPROVED** [review]

**Pull request merged**
